backend/app/api/dependencies.py

```
@@ -0,0 +1,77 @@
+    )
+
+    if http_credentials is None:
+        raise credentials_exception
+
+    token = http_credentials.credentials
+    if not token:
+        raise credentials_exception
+    try:
+        token_data = auth_service.get_data_from_token(token)
+    except Exception:
+        raise credentials_exception
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)